frontend/viewer/src/lib/components/stomp/stomp-guard.svelte.ts

```diff
@@ -0,0 +1,54 @@
+import {watch, type Getter} from 'runed';
+
+export class StompGuard<T> {
+
+  private _dirty = false;
+  private _value: T = $state<T>()!;
+
+  constructor(private readonly parentGetter: Getter<T>, private readonly parentSetter: (value: T) => void) {
+    this._value = parentGetter();
+    watch(parentGetter, (newParentValue) => {
+      if (newParentValue === this._value) return; // we probably updated the parent
+
+      if (this._dirty) {
+        // ignore and revert parent changes
+        parentSetter(this._value);
+      } else {
+        // accept parent changes
+        this._value = newParentValue;
+      }
+    });
+  }
+
+  get value(): T {
+    if (this._dirty) {
```

Comment:
because of `_dirty` determines which state values are subscribed to I think it needs to be a state value too. Otherwise `_value` is still being watched, even though dirty is false and now the getter should be watched instead. I wonder if I could write a failing test for that. Hum.